terraform/ratingservice/deploy_rating_service.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#------------------------------------------------------------------------------------------
+# Environment variables:
+# DELETE   - (optional) 1 or 0 to mark whether script should stop or deploy the app version
+# REGION   - (optional) defines GCP region where application has to be deployed
+# VERSION  - (optional) a version of the application to be deployed
+# INTERVAL - (optional) an interval in minutes between calls to recollect rating votes
+# DBHOST   - IP address of the Cloud SQL instance
+# DBNAME   - a name of the Postgres DB
+# DBUSER   - DB user
+# DBPWD    - DB user password
+#-------------------------------------------------------------------------------------------
+REGION=${REGION:-us-east1}
+VERSION=${VERSION:-prod}
+INTERVAL=${VERSION:-2}
+DELETE=${DELETE:-0}
+
+if [ $DELETE -eq 1 ]; then
+    gcloud app versions stop $VERSION --quiet # application is always deployed as default service
+    exit
+fi
```

Comment:
can we use terraform provider for app engine instead of this script?

Reply:
a terraform provider does not let to define the region. if we don't care about the region, i can replace `"local-exec"` provisioner with `google_app_engine_standard_app_version` provider.

Reply:
It sounds like this is probably out of scope of this PR since integration is coming later, but we may have to discuss this some more. One of our requirements is we want users to be able to run the install script multiple times to re-attach to existing projects. That's easy with terraform, since it keeps the current state of the system. If we have some parts created by terraform and some through scripts, it gets a bit complicated.

I'm not too familiar with terraform myself, but it's surprising that it would be flaky on AppEngine. That's one of the oldest GCP services

Reply:
I tried to use pure terraform resources (from Google provider) to install the service on AppEngine. This requires ~10 resources instead of one and does not work reliably. I agree that what can be done with terraform should work this way, but the fact is that you cannot do the same thing with 1-2 terraform resources. With that, I am ready to get it another try, maybe using Google beta provider to see if it works more reliable.